presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java

```diff
@@ -288,14 +288,26 @@ public void copyLogsToHostPath(Path hostPath)
         }
     }
 
-    public DockerContainer waitingForAll(WaitStrategy... strategies)
+    @Override
+    public DockerContainer waitingFor(WaitStrategy waitStrategy)
+    {
+        if (this.waitStrategy instanceof WaitAllStrategy) {
+            // add to existing wait all strategy
+            return super.waitingFor(((WaitAllStrategy) this.waitStrategy).withStrategy(waitStrategy));
+        }
+        else {
+            // merge existing strategies
+            return super.waitingFor(waitingForAll(this.waitStrategy, waitStrategy));
```

Comment:
This in turn disallows replacing the previous wait for.

Again https://github.com/prestosql/presto/pull/6094#discussion_r530668992

> i am concerned this will prevent us from reconfiguring a wait strategy along when reconfiguring the container. For example, there is Presto container with some Presto wait strategy using CLI and we now modify the container securing the frontend and we need to amend the wait strategy.